tests/debuginfo/basic-types-mut-globals.rs

```diff
@@ -49,58 +49,64 @@
 // gdbg-command:print 'basic_types_mut_globals::U64'
 // gdbr-command:print U64
 // gdb-check:$12 = 64
+// gdbg-command:print 'basic_types_mut_globals::F16'
+// gdbr-command:print F16
+// gdb-check:$13 = 1.5
 // gdbg-command:print 'basic_types_mut_globals::F32'
 // gdbr-command:print F32
-// gdb-check:$13 = 2.5
+// gdb-check:$14 = 2.5
 // gdbg-command:print 'basic_types_mut_globals::F64'
 // gdbr-command:print F64
-// gdb-check:$14 = 3.5
+// gdb-check:$15 = 3.5
 // gdb-command:continue
 
 // Check new values
 // gdbg-command:print 'basic_types_mut_globals'::B
 // gdbr-command:print B
-// gdb-check:$15 = true
+// gdb-check:$16 = true
 // gdbg-command:print 'basic_types_mut_globals'::I
 // gdbr-command:print I
-// gdb-check:$16 = 2
+// gdb-check:$17 = 2
 // gdbg-command:print/d 'basic_types_mut_globals'::C
 // gdbr-command:print C
-// gdbg-check:$17 = 102
-// gdbr-check:$17 = 102 'f'
+// gdbg-check:$18 = 102
+// gdbr-check:$18 = 102 'f'
 // gdbg-command:print/d 'basic_types_mut_globals'::I8
 // gdbr-command:print/d I8
-// gdb-check:$18 = 78
+// gdb-check:$19 = 78
 // gdbg-command:print 'basic_types_mut_globals'::I16
 // gdbr-command:print I16
-// gdb-check:$19 = -26
+// gdb-check:$20 = -26
 // gdbg-command:print 'basic_types_mut_globals'::I32
 // gdbr-command:print I32
-// gdb-check:$20 = -12
+// gdb-check:$21 = -12
 // gdbg-command:print 'basic_types_mut_globals'::I64
 // gdbr-command:print I64
-// gdb-check:$21 = -54
+// gdb-check:$22 = -54
 // gdbg-command:print 'basic_types_mut_globals'::U
 // gdbr-command:print U
-// gdb-check:$22 = 5
+// gdb-check:$23 = 5
 // gdbg-command:print/d 'basic_types_mut_globals'::U8
 // gdbr-command:print/d U8
-// gdb-check:$23 = 20
+// gdb-check:$24 = 20
 // gdbg-command:print 'basic_types_mut_globals'::U16
 // gdbr-command:print U16
-// gdb-check:$24 = 32
+// gdb-check:$25 = 32
 // gdbg-command:print 'basic_types_mut_globals'::U32
 // gdbr-command:print U32
-// gdb-check:$25 = 16
+// gdb-check:$26 = 16
 // gdbg-command:print 'basic_types_mut_globals'::U64
 // gdbr-command:print U64
-// gdb-check:$26 = 128
+// gdb-check:$27 = 128
+// gdbg-command:print 'basic_types_mut_globals'::F16
+// gdbr-command:print F16
+// gdb-check:$28 = 2.25
 // gdbg-command:print 'basic_types_mut_globals'::F32
 // gdbr-command:print F32
-// gdb-check:$27 = 5.75
+// gdb-check:$29 = 5.75
 // gdbg-command:print 'basic_types_mut_globals'::F64
 // gdbr-command:print F64
-// gdb-check:$28 = 9.25
+// gdb-check:$30 = 9.25
 
 #![allow(unused_variables)]
 #![feature(omit_gdb_pretty_printer_section)]
```

Comment:
Needs a `#![feature(f16)]`.

If you're up for it you can try to remove the `//@ ignore-gdb` directive at the top, since the very similar `basic-types-globals.rs` test seems to work just fine.